x-pack/plugins/security_solution/public/detections/components/rules/step_rule_actions/schema.tsx

```diff
@@ -13,42 +13,47 @@ import {
   AlertAction,
   ActionTypeRegistryContract,
 } from '../../../../../../triggers_actions_ui/public';
-import { FormSchema, ValidationFunc, ERROR_CODE } from '../../../../shared_imports';
+import {
+  FormSchema,
+  ValidationFunc,
+  ERROR_CODE,
+  ValidationError,
+} from '../../../../shared_imports';
 import { ActionsStepRule } from '../../../pages/detection_engine/rules/types';
 import * as I18n from './translations';
 import { isUuid, getActionTypeName, validateMustache, validateActionParams } from './utils';
 
-export const validateSingleAction = (
+export const validateSingleAction = async (
   actionItem: AlertAction,
   actionTypeRegistry: ActionTypeRegistryContract
-): string[] => {
+): Promise<string[]> => {
   if (!isUuid(actionItem.id)) {
     return [I18n.NO_CONNECTOR_SELECTED];
   }
 
-  const actionParamsErrors = validateActionParams(actionItem, actionTypeRegistry);
+  const actionParamsErrors = await validateActionParams(actionItem, actionTypeRegistry);
   const mustacheErrors = validateMustache(actionItem.params);
 
   return [...actionParamsErrors, ...mustacheErrors];
 };
 
-export const validateRuleActionsField = (actionTypeRegistry: ActionTypeRegistryContract) => (
+export const validateRuleActionsField = (actionTypeRegistry: ActionTypeRegistryContract) => async (
```

Comment:
Hey! Maybe it is best to use a simple for loop. I think is more readable and more performant. The `async` `reduce` is difficult for me to understand it. Example:

```
const errors = []
for (const actionItem of value) {
    const errorsArray = await validateSingleAction(actionItem, actionTypeRegistry);

    if (errorsArray.length) {
      const actionTypeName = getActionTypeName(actionItem.actionTypeId);
      const errorsListItems = errorsArray.map((error) => `*   ${error}\n`);
	}

	errors.push(`\n**${actionTypeName}:**\n${errorsListItems.join('')}`)
}

return acc
```

Reply:
fixed in [commit](https://github.com/elastic/kibana/pull/100525/commits/fccdb714941561d6d0bd79ce41c8fe610fcfbe7f)